docs/USING_PUBLIC_CLOUDS.rst

```diff
@@ -0,0 +1,374 @@
+There are a few configurable options here:
+
+- :code:`type` : set to :code:`aws_s3` to indicate this location constraint is
+  writing data to AWS S3;
+- :code:`legacyAwsBehavior` : set to :code:`true` to indicate this region should
+  behave like AWS S3 :code:`us-east-1` region, set to :code:`false` to indicate
+  this region should behave like any other AWS S3 region;
+- :code:`bucketName` : set to an *existing bucket* in your AWS S3 Account; this
+  is the bucket in which your data will be stored for this location constraint;
+- :code:`awsEndpoint` : set to your bucket's endpoint, usually :code:`s3.amazonaws.com`;
+- :code:`bucketMatch` : set to :code:`true` if you want your object name to be the
+  same in your local bucket and your AWS S3 bucket; set to :code:`false` if you
+  want your object name to be of the form :code:`{{localBucketName}}/{{objectname}}`
+  in your AWS S3 hosted bucket;
+- :code:`credentialsProfile` and :code:`credentials` are two ways to provide
+  your AWS S3 credentials for that bucket, *use only one of them* :
+
+  - :code:`credentialsProfile` : set to the profile name allowing you to access
+    your AWS S3 bucket from your :code:`~/.aws/credentials` file;
+  - :code:`credentials` : set the two fields inside the object (:code:`accessKey`
+    and :code:`secretKey`) to their respective values from your AWS credentials.
+
+.. code:: json
+
+    (...)
+    "aws-test": {
+        "type": "aws_s3",
+        "legacyAwsBehavior": true,
+        "details": {
+            "awsEndpoint": "s3.amazonaws.com",
+            "bucketName": "zenkobucket",
+            "bucketMatch": true,
+            "credentialsProfile": "zenko"
+        }
+    },
+    (...)
+
+.. code:: json
+
+    (...)
+    "aws-test": {
+        "type": "aws_s3",
+        "legacyAwsBehavior": true,
+        "details": {
+            "awsEndpoint": "s3.amazonaws.com",
+            "bucketName": "zenkobucket",
+            "bucketMatch": true,
+            "credentials": {
+                "accessKey": "WHDBFKILOSDDVF78NPMQ",
+                "secretKey": "87hdfGCvDS+YYzefKLnjjZEYstOIuIjs/2X72eET"
+            }
+        }
+    },
+    (...)
+
+.. WARNING::
+   If you set :code:`bucketMatch` to :code:`true`, we strongly advise that you
+   only have one local bucket per AWS S3 location.
+   Without :code:`bucketMatch` set to :code:`false`, your object names in your
+   AWS S3 bucket will not be prefixed with your Cloud Server bucket name. This
+   means that if you put an object :code:`foo` to your CloudServer bucket
+   :code:`zenko1` and you then put a different :code:`foo` to your CloudServer
+   bucket :code:`zenko2` and both :code:`zenko1` and :code:`zenko2` point to the
+   same AWS bucket, the second :code:`foo` will overwrite the first :code:`foo`.
+
+config.json
+^^^^^^^^^^^
+
+.. IMPORTANT::
+   You only need to follow this section if you want to define your new location
+   :code:`aws-test` as the default for AWS S3
```

Comment:
as the default for AWS S3? as the default for an endpoint...

Reply:
Well, in the context of this guide, it's for s3.amazonaws.com, which I shortened to AWS S3. I think it might be easier to comprehend that with the example. I agree though we could document that config.json should only be modified to add defaults for endpoints. Maybe in a more generic / less "how-to" oriented file?

Reply:
I would say "as the default for a specific endpoint (e.g., 'zenkotos3.com')"

And then use zenkotos3.com as the example below and remove references to default AWS S3 location. I just find it really confusing to say we have a default AWS S3 location.

Reply:
Edited, and moved that section to the end as it is not specific to S3